R/bnma_analysis.R

```diff
@@ -267,6 +267,90 @@ GetReferenceOutcome <- function(data, treatment_ids, outcome_type, observed, mod
 
 
 
+#' Calculate the confidence regions within direct evidence for the baseline risk model.
+#'
+#' @param model_output Return from `BaselineRiskModelOutput()`.
+#'
+#' @return list of confidence region objects and confidence interval objects.
+#' Regions cover treatments with a non-zero covariate range of direct contributions,
+#' intervals cover treatments with a single covariate value from direct contributions.
+#' Any treatment with no direct contributions will not be present in either list.
+#' Each is a list of data frames for each treatment name. Each data frame contains 3 columns:
+#' - cov_value: The covariate value at which the confidence region is calculated.
+#' - lower: the 2.5% quantile.
+#' - upper: the 97.5% quantile.
+#' Each data frame in "regions" contains 11 rows creating a 10-polygon region.
+#' Each data frame in "intervals" contains a single row at the covariate value of that single contribution.
+CalculateConfidenceRegionsBnma <- function(model_output) {
+  
+  mtc_results <- model_output$mtcResults
+  treatments <- mtc_results$network$Treat.order
+  
+  confidence_regions <- list()
+  confidence_intervals <- list()
+  
+  for (treatment_name in model_output$comparator_names) {
+    parameter_name <- glue::glue("d[", which(treatment_name == unname(treatments)), "]")
```

Comment:
I think that's not the intended way to use glue. The usual way would be:
`glue::glue("d[{which(treatment_name == unname(treatments))}]")`

Optional if you want to change it.